### PR TITLE
[PATCH v5] api: pktio: add packet input queue size configuration option

### PR DESCRIPTION
**Pull request by MatiasElo**

### CODE REVIEW

File: include/odp/api/spec/packet_io_types.h [hunk 1]
```diff
@@ -265,6 +265,16 @@ typedef struct odp_pktin_queue_param_t {
...
 
+	/** Input queue size array
+	  *
+	  * An array containing queue sizes for each 'num_queues' input queues.
+	  * The value of zero means implementation specific default size.
+	  * Nonzero values must be between 'min_input_queue_size' and
+	  * 'max_input_queue_size' capabilities. The implementation may
+	  * round-up given values. The default value is zero.
+	  */
+	uint32_t queue_size[ODP_PKTIN_MAX_QUEUES];
```
Comment (psavol) [inline include/odp/api/spec/packet_io_types.h:276]: We should mention that this applies to only DIRECT mode.

**Comment by nithind1988** [comment]
This series looks good to me if Petri's comment is addressed i.e  need to explictly say that it is for PKTIN_DIRECT mode.

File: include/odp/api/spec/packet_io_types.h (hunk 1 above)
Comment (MatiasElo) [reply include/odp/api/spec/packet_io_types.h:276]: Added mention in V3.

**Comment by MatiasElo** [comment]
V3:
* Document that input queue size configuration is only for ODP_PKTIN_MODE_DIRECT mode
* Added implementation and tests

**Comment by nithind1988** [comment]
Looks good to me.

Reviewed-by: Nithin Dabilpuram <ndabilpuram@marvell.com>

File: platform/linux-generic/pktio/dpdk.c [hunk 2]
```diff
@@ -1493,21 +1489,50 @@ static void prepare_rss_conf(pktio_entry_t *pktio_entry,
...
+			if (dev_info.rx_desc_lim.nb_align)
+				num_rx_desc = RTE_ALIGN_MUL_CEIL(num_rx_desc,
+								 dev_info.rx_desc_lim.nb_align);
+		}
+
+		if (num_rx_desc < dev_info.rx_desc_lim.nb_min ||
+		    num_rx_desc > dev_info.rx_desc_lim.nb_max ||
+		    num_rx_desc % dev_info.rx_desc_lim.nb_align) {
+			_ODP_ERR("DPDK: invalid number of RX descriptors\n");
+			return -1;
```
Comment (psavol) [inline platform/linux-generic/pktio/dpdk.c:1532]: For easier debug, this could print num_rx_desc and the index.

Comment (MatiasElo) [reply platform/linux-generic/pktio/dpdk.c:1533]: Fixed in V4.

**Pull request merged**